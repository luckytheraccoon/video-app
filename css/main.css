html{overflow-x:hidden}body,html{margin:0;padding:0;font-family:'Varela Round',sans-serif;font-size:14px;width:100%;height:100%}#root{margin:0 auto;max-width:1000px;min-width:260px;height:100%}#root>div{width:100%;height:100%}a{cursor:pointer;color:#317795}a:link,a:visited{text-decoration:none!important;font-weight:400}a:focus,a:hover{color:#23527c}.div-loading{padding-top:10px;text-align:center}.div-main{display:grid;width:100%;height:100%}.div-maincontent-wrapper{align-self:start;margin:10px;grid-area:maincontent}.div-search-form{margin-bottom:10px;flex:1 100%;justify-content:space-between;flex-wrap:wrap;align-self:start}.div-search-form>input.search-box{flex-grow:1;margin-right:2px}.div-search-form>input.submit-button{flex-shrink:1}.div-user-block{text-align:center}.div-user-block>.div-user-photo{font-size:5em;line-height:.1em;padding:10px}.div-main-menu{background:#f1f1f1;z-index:1;flex-direction:column;justify-content:space-between;min-width:150px}.div-main-menu>div{flex-shrink:1}.div-main-menu>.div-main-menu-menu{flex-grow:1;padding:0 5px 5px 5px}.div-main-menu>.div-main-menu-menu>button{width:100%;margin-top:5px}.div-main-menu>.div-main-menu-logout{padding:5px}.div-main-menu>.div-main-menu-logout>button{width:100%}.elementHidden{display:none}.div-content-list,.div-main-menu.elementVisible,.div-search-form.elementVisible,.div-side-content-list{display:flex}.div-content-list{flex-flow:wrap}.div-content-detail,.div-content-list,.div-side-content-list{position:relative}.div-content-detail>div>div,.div-content-list>div>div,.div-side-content-list>div>div{margin:0 0 10px 0}.div-content-detail>div>div.title,.div-content-list>div>div.title,.div-side-content-list>div>div.title{font-size:.9em;margin:0}.div-content-detail>div>div.description,.div-content-list>div>div.description,.div-side-content-list>div>div.description{font-size:.8em}.div-side-content-list{flex-direction:column;padding-left:10px;flex:1 20%}.div-content-detail>div>div>div.title{margin-top:10px;font-size:1.2em;font-weight:700}.div-footer{grid-area:footer;display:flex;justify-content:space-between;flex-wrap:wrap;background-color:#f1f1f1;padding:25px;color:#888}.div-footer>div{margin:5px;flex-shrink:1}.div-footer>div:last-child{text-align:right;flex-grow:1}.video-iframe{position:absolute}.thumbnail-large>a>img{width:100%;max-width:275px;cursor:pointer}.thumbnail-medium>a>img{width:100%;cursor:pointer}.glyphicon{cursor:pointer}a>.glyphicon{color:#fff}.go-back-button,.more-results-button,.more-results-button-no-click{width:100%!important;padding:10px!important;background:#f1f1f1;border:1px solid #ccc;margin-bottom:10px;text-align:center;cursor:pointer;color:#000}.go-back-button>.glyphicon,.more-results-button-no-click>.glyphicon,.more-results-button>.glyphicon{color:#000}.go-back-button:hover,.more-results-button-no-click:hover,.more-results-button:hover{border:1px solid #999;background:#f5f5f5}.more-results-button-no-click:hover{background:#f1f1f1;border:1px solid #f1f1f1}@media (min-width:768px) and (max-width:991px){.div-main{grid-template-rows:[maincontent] 9fr [footer] 1fr;grid-template-areas:"maincontent-left maincontent" "footer footer";grid-template-columns:[left] 1fr [right] 9fr}.div-maincontent-wrapper{display:flex;flex-flow:row wrap}.div-side-content-list{padding:10px 0}.div-side-content-list .thumbnail-medium{float:left;margin-right:10px}.div-content-detail,.div-content-list{flex:1 100%}.div-content-detail.elementHidden,.div-content-list.elementHidden{display:flex}.div-content-detail>div>.thumbnail-large,.div-content-list>div>.thumbnail-large{float:left;margin-right:10px}.div-content-detail>div>div{flex:1 100%}.div-content-list>div{flex:1 100%}.div-search-form{grid-area:searchbar}.div-search-form.elementHidden{display:flex}.div-main-menu{grid-area:maincontent-left}.div-main-menu.elementHidden{display:flex}.div-mobile-top-menu-center,.div-mobile-top-menu-left,.div-mobile-top-menu-right{display:none}}@media (min-width:992px){.div-main{grid-template-rows:[maincontent] 9fr [footer] 1fr;grid-template-areas:"maincontent-left maincontent" "footer footer";grid-template-columns:[left] 1fr [right] 9fr}.div-maincontent-wrapper{display:flex;flex-flow:row wrap}.div-content-detail,.div-content-list{flex:1 80%}.div-content-detail.elementHidden,.div-content-list.elementHidden{display:flex}.div-content-detail>div>div{flex:1 100%}.div-content-list>div{padding:0 5px;width:33.3%}.div-search-form{grid-area:searchbar}.div-search-form.elementHidden{display:flex}.div-main-menu{grid-area:maincontent-left}.div-main-menu.elementHidden{display:flex}.div-mobile-top-menu-center,.div-mobile-top-menu-left,.div-mobile-top-menu-right{display:none}}@media (max-width:767px){.div-main{grid-template-rows:[header] 60px [maincontent] 9fr [footer] 1fr;grid-template-areas:"mobile-top-menu-left mobile-top-menu-center mobile-top-menu-right" "maincontent maincontent maincontent" "footer footer footer";grid-template-columns:[left] 2fr [center] 6fr [right] 2fr}.go-back-button{display:none}.div-side-content-list{padding:10px 0;flex-direction:column;flex:1 100%}.div-side-content-list .thumbnail-medium{width:40%;float:left;margin-right:10px}.div-content-detail,.div-content-list{flex:1 100%}.div-content-detail>div,.div-content-list>div{flex:1 100%}.div-content-detail>div>.thumbnail-large,.div-content-list>div>.thumbnail-large{width:40%;float:left;margin-right:10px}.div-content-detail>div>.description,.div-content-detail>div>.title,.div-content-list>div>.description,.div-content-list>div>.title,.div-side-content-list>div>.description,.div-side-content-list>div>.title{width:55%;float:left}.div-main-menu{grid-area:maincontent;padding:10px}.div-mobile-top-menu-left{grid-area:mobile-top-menu-left;text-align:left;padding:10px 0 4px 14px;font-size:2em;background-color:#b2b200;position:relative;color:#fff}.div-mobile-top-menu-center{grid-area:mobile-top-menu-center;text-align:center;padding:10px 0 4px 0;font-size:2em;background-color:#b2b200;position:relative;color:#fff}.div-mobile-top-menu-right{grid-area:mobile-top-menu-right;text-align:right;padding:10px 14px 4px 0;font-size:2em;background-color:#b2b200;position:relative;color:#fff}}@media (max-width:390px){.div-main{grid-template-rows:[header] 40px [maincontent] 9fr [footer] 1fr}.div-mobile-top-menu-center,.div-mobile-top-menu-left,.div-mobile-top-menu-right{font-size:1.2em}}