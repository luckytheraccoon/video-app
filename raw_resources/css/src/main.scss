@mixin full-span() {
	width:100%;
	height:100%;
}

@function str-replace($string, $search, $replace: '') {
  $index: str-index($string, $search);

  @if $index {
    @return str-slice($string, 1, $index - 1) + $replace + str-replace(str-slice($string, $index + str-length($search)), $search, $replace);
  }

  @return $string;
}

@mixin prefix($declarations, $prefixes: ()) {
  @each $property, $value in $declarations {
    @each $prefix in $prefixes {
      @if type_of($value)==string  {
        $value: str-replace($value, '-', ',');
        $value: str-replace($value, '_', ' ');
      } 
      #{'-' + $prefix + '-' + $property}: $value;
    }

    // Output standard non-prefixed declaration
    @if type_of($value)==string  {
      $value: str-replace($value, '-', ',');
      $value: str-replace($value, '_', ' ');
    }
    #{$property}: $value;
  }
}
html {
	overflow-x: hidden;
}
html, body {
	margin:0;
	padding:0;
	font: {
		family: 'Varela Round', sans-serif;
		size:14px;
	}
	@include full-span();
}
#root {
	margin:0 auto;
	max-width:1000px;
	min-width:260px;
	height:100%;
	&>div {
		@include full-span();
	}
}
a {
	cursor:pointer;
	color: #317795;
	&:link, &:visited {
	    text-decoration: none !important;
	    font-weight: normal;
	}

	&:hover, &:focus {
	    color: #23527c;
	}

}

@mixin div-main-grid() {
	display: grid;
	@include full-span();
}

@mixin div-header() {
	background-color:#B2B200;
	position: relative;
	color: white;
}

.div-loading {
	padding-top:10px;
	text-align: center;
}

.div-main {
	@include div-main-grid();
}

.div-maincontent-wrapper {
	align-self: start;
	margin: 10px;
	grid-area: maincontent;
}

.div-search-form {
	margin-bottom: 10px;
    flex: 1 100%;
	justify-content: space-between;
	flex-wrap: wrap;
	align-self: start;
	&>input.search-box {
		flex-grow: 1;
		margin-right:2px;
	}
	&>input.submit-button {
		flex-shrink: 1;
	}
}

.div-user-block {
	text-align:center;
	&>.div-user-photo {
		font-size: 5em;
		line-height: 0.1em;
		padding: 10px;
	}
}

.div-main-menu {
	background: #f1f1f1;
    z-index: 1;
    flex-direction: column;
    justify-content: space-between;
	min-width: 150px;
	&>div {
		flex-shrink: 1;
	}

	&>.div-main-menu-menu {
		flex-grow: 1;
		padding:0 5px 5px 5px;
		&>button {
			width: 100%;
			margin-top:5px;
		}
	}

	&>.div-main-menu-logout {
		padding:5px;
		&>button {
			width: 100%;
		}
	}
}

.elementHidden {
	display:none; 
}
.div-side-content-list,
.div-content-list,
.div-search-form.elementVisible,
.div-main-menu.elementVisible {
	display: flex;
}

.div-content-list {
	flex-flow:wrap;
}

.div-side-content-list,
.div-content-detail,
.div-content-list {
	position:relative;
	&>div>div {
		margin:0 0 10px 0;
	}

	&>div>div.title {
		font-size:0.9em;
		margin:0;
	}

	&>div>div.description {
		font-size:0.8em;
	}
}

.div-side-content-list {
	flex-direction:column;
	padding-left: 10px;
	flex: 1 20%;
}

.div-content-detail {
	&>div>div>div.title {
		margin-top:10px;
		font-size:1.2em;
		font-weight: bold;
	}
}
.div-footer {
	grid-area: footer;
	display: flex;
	justify-content: space-between;
	flex-wrap: wrap;
	background-color:#f1f1f1;
	padding:25px;
	color:#888;
	&>div {
		margin:5px;
		flex-shrink:1;
	}
	&>div:last-child {
		text-align: right;
		flex-grow:1;
	}
}
.video-iframe {
	position:absolute;
}

.thumbnail-large>a>img {
	width:100%;
	max-width: 275px;
	cursor: pointer;
}

.thumbnail-medium>a>img {
	width:100%;
	cursor: pointer;
}
.glyphicon {
	cursor: pointer;
}
a>.glyphicon {
	color:white;
}


.more-results-button,
.go-back-button,
.more-results-button-no-click {
	width: 100%!important;
	padding: 10px!important;
	background: #f1f1f1;
	border:1px solid #CCC;
	margin-bottom: 10px;
	text-align: center;
	cursor: pointer;
	color:black;

	&>.glyphicon {
		color:black;
	} 

	&:hover {
		border:1px solid #999;
		background: #f5f5f5;
	}

}

.more-results-button-no-click {
	&:hover {
		background: #f1f1f1;
		border:1px solid #f1f1f1;
	}
}

/*
	Large Tablets / Laptops
*/
@media (min-width:768px) and (max-width:991px) {
    .div-main {
		grid-template: {
			rows: [maincontent] 9fr [footer] 1fr;
			areas: 
			"maincontent-left maincontent"
			"footer footer";
			columns: [left] 1fr [right] 9fr;
		}
	}
	.div-maincontent-wrapper {
		display:flex;
		flex-flow: row wrap;
	}
	.div-side-content-list {
		padding:10px 0;
		& .thumbnail-medium {
			float:left;
			margin-right:10px;
		}
	}
	.div-content-list, 
	.div-content-detail {
		flex: 1 100%;

		&.elementHidden {
			display: flex;
		}
		
		&>div>.thumbnail-large {
			float:left;
			margin-right:10px;
		}
	}
	.div-content-detail {
		&>div>div {
			flex: 1 100%;
		}
	}
	.div-content-list {
		&>div {
			flex: 1 100%;
		}
	}
	.div-search-form {
		grid-area: searchbar;
		&.elementHidden {
			display:flex;
		}
	}

	.div-main-menu {
		grid-area: maincontent-left;  
		&.elementHidden {
			display:flex;
		}
	}

	.div-mobile-top-menu-left,
	.div-mobile-top-menu-center,
	.div-mobile-top-menu-right {
		display:none
	}
}
/*
	Laptops / Desktops
*/
@media (min-width:992px)  {
    .div-main {
		grid-template: {
			rows: [maincontent] 9fr [footer] 1fr;
			areas: 
			"maincontent-left maincontent"
			"footer footer";
			columns: [left] 1fr [right] 9fr;
		}
	}
	.div-maincontent-wrapper {
		display:flex;
		flex-flow: row wrap;
	}
	.div-content-detail,
	.div-content-list {
    	flex: 1 80%;
		&.elementHidden {
			display: flex;
		}
	}
	.div-content-detail {
		&>div>div {
			flex: 1 100%;
		}
	}
	.div-content-list {
		&>div {
			padding: 0 5px;
			width: 33.3%;
		}
	}

	.div-search-form {
		grid-area: searchbar;
		&.elementHidden {
			display:flex;
		}
	}

	.div-main-menu {
		grid-area: maincontent-left;  
		&.elementHidden {
			display:flex;
		}
	}

	.div-mobile-top-menu-left,
	.div-mobile-top-menu-center,
	.div-mobile-top-menu-right {
		display:none
	}
}

/*
	Small Tablets
*/
@media (max-width:767px) {
    .div-main {
		grid-template: {
			rows: [header] 60px [maincontent] 9fr [footer] 1fr;
			areas: 
			"mobile-top-menu-left mobile-top-menu-center mobile-top-menu-right"
			"maincontent maincontent maincontent"
			"footer footer footer";
			columns: [left] 2fr [center] 6fr [right] 2fr;
		}
	}

	.go-back-button {
		display:none;
	}
	
	.div-side-content-list {
		padding:10px 0;
		flex-direction:column;
    	flex: 1 100%;
		& .thumbnail-medium {
			width: 40%;
			float:left;
			margin-right:10px;
		}
	}
	
	.div-content-detail,
	.div-content-list {
    	flex: 1 100%;
		
		&>div {
			flex: 1 100%;
		}
		&>div>.thumbnail-large {
			width: 40%;
			float:left;
			margin-right:10px;
		}
	}

	.div-side-content-list,
	.div-content-detail,
	.div-content-list {
		&>div>.title,
		&>div>.description {
			width: 55%;
			float:left;
		}
	}

	.div-main-menu {
		grid-area: maincontent;    
		padding: 10px;
	}

	@mixin div-mobile-top-menu() {
		font-size:2em;
		@include div-header();
	}
	.div-mobile-top-menu-left {
		grid-area: mobile-top-menu-left;
		text-align:left;
		padding: 10px 0 4px 14px;
		@include div-mobile-top-menu();
	}
	.div-mobile-top-menu-center {
		grid-area: mobile-top-menu-center;
		text-align:center;
		padding: 10px 0 4px 0;
		@include div-mobile-top-menu();
	}
	.div-mobile-top-menu-right {
		grid-area: mobile-top-menu-right;
		text-align:right;
		padding: 10px 14px 4px 0;
		@include div-mobile-top-menu();
	}
}

/*
	Smartphones
*/
@media (max-width:390px) {
	
    .div-main {
		grid-template-rows: [header] 40px [maincontent] 9fr [footer] 1fr;
	}
	.div-mobile-top-menu-left,
	.div-mobile-top-menu-center,
	.div-mobile-top-menu-right {
		font-size:1.2em;
	}
}