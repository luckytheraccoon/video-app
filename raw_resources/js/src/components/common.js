import React from "react";
//the mainContent module is responsible for loading up specific a content item, or a list of content items
import fetchContent from "../data/mainContent";
/**
 * I prefer to keep some general components stashed in a single so I don't overuse imports too much.
 */

/**
 * This renders a content item's details when being listed. 
 * 
 * Uses a switch to decide on particular details before actually rendering.
 * 
 * @param {object} props - The component properties from within the react context.
 */
function ContentItemList(props) {

    let content = props.contentData;
    let thumbnailClass;
    
    //its possible to limit the amount of characters on a title or description this way
    let title = content.title.substring(0,35) + "...";
    let description = content.description.substring(0,55) + "...";
    
    switch(props.pageId) {
        case "video-list":
            thumbnailClass = "thumbnail-large";
            break;
        case "side-video-list":
            thumbnailClass = "thumbnail-medium";
            break;
    }
     
    return (
        <div id={"content_"+content.index}>
            <div className={thumbnailClass}>
                <DetailViewLink contentId={content.id}>
                    <img src={content.thumbnail} />
                </DetailViewLink>
            </div>
            <div className="title">
                <DetailViewLink contentId={content.id}>
                    {title}
                </DetailViewLink>
            </div>            
            <div className="description">
                {description}
            </div>
        </div>
    );
}

/** 
* This is cheating react, i know... but i feel like performance-wise this is a better idea instead of 
* rellying on state changes on window-resize...!
*/
var iframeComponentId;
window.onresize = function() {
    if(typeof iframeComponentId != "undefined") {
        let videoIframe = document.getElementById(iframeComponentId);
        videoIframe.width = videoIframe.parentElement.parentElement.offsetWidth;
        videoIframe.height = Math.floor(videoIframe.width / 1.77);
        videoIframe.parentElement.style.height = videoIframe.height + "px";
    }
}
/**
 * A component to render an iframe with an external video url.
 */
class VideoIframe extends React.PureComponent {

    constructor(props) {
        super(props);

        //multiple details related to the iframe HTML element are defined here
        this.compId = "video-iframe";
        this.compClass = this.compId;
        this.serverProtocol = "https:";
        this.startingWidth = 0;
        this.startingHeight = 0;
        this.frameBorder = 0;
    }

    /**
     * This feeds the resize window method a iframe id to resize based on window size.
     */
    componentWillMount() {
        iframeComponentId = this.compId;
    }

    /**
     * This is so the resize method doesnt error out when an iframe isnt present.
     */
    componentWillUnmount() {
        iframeComponentId = null;
    }

    /**
     * Call the resize method as soon as an iframe is created, this way it shows up correctly sized immediatly.
     */
    componentDidMount() {
        window.onresize();
    }

    render() {
        return <iframe 
                    id={this.compId} 
                    className={this.compClass} 
                    src={this.serverProtocol+this.props.videoUrl} 
                    width={this.startingWidth} 
                    height={this.startingHeight} 
                    frameBorder={this.frameBorder}>
               </iframe>;
    }
}
/**
 * A simple component that renders the detail view of a item.
 * 
 * @param {object} props - The component properties from within the react context.
 */
function ContentItemDetail(props) {

    let content = props.contentData;
    
    return (
        <div>
            <div className="div-iframe-container">
                <VideoIframe videoUrl={content.videoUrl} />
            </div>
            <div className="title">
                {content.title}
            </div>
            <div className="description">
                {content.description}
            </div>
        </div>
    );
}
/**
 * A simple component that renders a simple "load more" button. 
 * 
 * Its not very customizable, this one is for a single purpose.
 */
class LoadMoreButton extends React.PureComponent {

    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }
    
    handleClick() {
        //if the clickAction came through as null, then if we click the button, nothing should happen
        //otherwise, run the action
        if(this.props.clickAction) {
            this.props.clickAction();
        }
    }

    render() {

        //based on wether we can get more results or not, we define the button label
        let label = "Load More Videos";
        let className = "more-results-button";

        if(this.props.clickAction === null) {
            label = "All Videos Loaded" ;
            className = "more-results-button-no-click";
        }

        return <div className={className} onClick={this.handleClick}>{label}</div>;
    }
}

/**
 * This is a simple icon that can have a callback attached to it for all your needs.
 * Granted, the callback system right now is not fully developed, it could be more robust, as you can see the callback takes a pre-defined stack of parameters.
 * To-Do: Make use of spread or some other way to allow for truly flexible callbacks.
 */
class Glyphicon extends React.PureComponent {

    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }
    
    handleClick() {
        if(this.props.clickAction) {
            this.props.clickAction(this.props.target, this.props.pageTitle, this.props.contentId);
        }
    }

    render() {
        return <span onClick={this.handleClick} className={"glyphicon glyphicon-" + this.props.iconSuffix} >{this.props.children}</span>;
    }
}
/**
 * Just a link that will send the user to the video list.
 * 
 * @param {object} props - The component properties from within the react context.
 */
function ListViewLink(props) {
    return <a href="?videos">{props.children}</a>;
}

/**
 * Just a link that will send the user to the video detail.
 * 
 * @param {object} props - The component properties from within the react context.
 */
function DetailViewLink(props) {
    return <a href={"?video="+props.contentId}>{props.children}</a>;
}



/**
 * Video detail page.
 * 
 * Sets the window title according to what video is being viewed.
 */
class VideoDetail extends React.PureComponent {

    constructor(props) {
        super(props);
        this.contentData;
    }

    componentWillMount() {
        this.contentData = fetchContent(this.props.contentArray, this.props.contentId);
        this.props.setTitleAction(this.contentData.title);
    }

    render() {
        return(
            <div className="div-content-detail">
                <ListViewLink>
                    <div className="go-back-button">
                        <Glyphicon key="triangle-left" iconSuffix="triangle-left"  />
                        Back to Video List
                    </div>
                </ListViewLink>
                <div>
                    <ContentItemDetail contentData={this.contentData}  />
                </div>
            </div>
        );
    }
}

/**
 * Video list page.
 * @param {object} props - The component properties from within the react context.
 */
function VideoList(props) {

    //this will return null if we can't actually load more
    let loadMoreAction = props.loadMoreAction();

    let pageComponents = [];

    //get each content item and call a component to render it
    for(var key in props.contentArray) {
        if(props.contentArray.hasOwnProperty(key)) {
            pageComponents.push(
                <ContentItemList 
                    pageId="video-list" 
                    key={key} 
                    contentData={fetchContent(props.contentArray, key)} 
                />
            );
        }
    }

    return(
        <div className="div-content-list">
            {pageComponents}
            <LoadMoreButton clickAction={loadMoreAction} />
        </div>
    );
}


/**
 * Side Video list block, not an actual page, just a block that is embedded within other pages.
 * @param {object} props - The component properties from within the react context.
 */
function SideVideoList(props) {

    //this will return null if we can't actually load more
    let loadMoreAction = props.loadMoreAction();
    
    let pageComponents = [];
    
    //get each content item and call a component to render it, dont display the current item in detail view
    for(var key in props.contentArray) {
        if(props.contentArray.hasOwnProperty(key)) {
            if(key !== props.contentId) {
                pageComponents.push(<ContentItemList pageId="side-video-list" key={key} contentData={fetchContent(props.contentArray, key)} />);
            }
        }
    }

    return(
        <div className="div-side-content-list">
            {pageComponents}
            <LoadMoreButton clickAction={loadMoreAction} />
        </div>
    );
}

export { VideoList,  Glyphicon, ListViewLink, VideoDetail, SideVideoList };