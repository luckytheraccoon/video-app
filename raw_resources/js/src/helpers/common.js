/**
 * A helper method that builds a URL based on some parameters
 * 
 * @param {string} requestSearchTerm - The search term to send to the server.
 * @param {string} reqPage - The page the user is requesting.
 * @param {int} itemLimit - The item limit to return.
 * @param {int} itemId - A specific item ID.
 */
function buildApiUrl(requestSearchTerm, reqPage, itemLimit, itemId) {
    let apiUrl = "http://api.joinolx.com/api/v1/videos";
    let apiToken = "olx-token=0a896ba98dacbd3f438050fc80d2a9e359059276";

    if(itemId) {
        return apiUrl+"/"+itemId+"?"+apiToken;
    }

    apiUrl += requestSearchTerm ? "/search?q="+ requestSearchTerm + "&" : "?" ;
    let apiPage = "&page="+reqPage;
    let apiLimit = "&limit="+itemLimit;
    return apiUrl+apiToken+apiPage+apiLimit;
}
/**
 * In order to implement a very basic URL router, this method helps with retrieving URL data.
 * 
 * @param {string} paramName - Parameter name to get from the URL.
 */
function getUrlParameter(paramName) {
    let url = window.location.href;
    paramName = paramName.replace(/[\[\]]/g, "\\$&");
    var regex = new RegExp("[?&]" + paramName + "(=([^&#]*)|&|#|$)"),results = regex.exec(url);
    if (!results) return null;
    if (!results[2]) return '';
    return decodeURIComponent(results[2].replace(/\+/g, " "));
}

export { getUrlParameter, buildApiUrl };