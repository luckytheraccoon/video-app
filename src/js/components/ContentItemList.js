import React from "react";

import DetailViewLink from "components/DetailViewLink"; 

/**
 * This renders a content item's details when being listed. 
 * 
 * Uses a switch to decide on particular details before actually rendering.
 * 
 * @param {object} props - The component properties from within the react context.
 */
export default function ContentItemList(props) {

    let content = props.contentData;
    let thumbnailClass;

    //its possible to limit the amount of characters on a title or description this way
    let title = content.title.substring(0, 35) + "...";
    let description = content.description.substring(0, 55) + "...";

    switch (props.pageId) {
        case "video-list":
            thumbnailClass = "thumbnail-large";
            break;
        case "side-video-list":
            thumbnailClass = "thumbnail-medium";
            break;
    }

    return (
        <div id={"content_" + content.index}>
            <div className={thumbnailClass}>
                <DetailViewLink contentId={content._id}>
                    <img src={content.thumbnailUrl} />
                </DetailViewLink>
            </div>
            <div className="title">
                <DetailViewLink contentId={content._id}>
                    {title}
                </DetailViewLink>
            </div>
            <div className="description">
                {description}
            </div>
        </div>
    );
} 