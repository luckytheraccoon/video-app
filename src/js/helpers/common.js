module.exports = {

    /**
     * A helper method that builds a URL based on some parameters
     * 
     * @param {string} requestSearchTerm - The search term to send to the server.
     * @param {string} reqPage - The page the user is requesting.
     * @param {int} itemLimit - The item limit to return.
     * @param {int} itemId - A specific item ID.
     */
    buildApiUrl: function (query) {
        return "http://localhost:8080/" + query;
    },
    /**
     * In order to implement a very basic URL router, this method helps with retrieving URL data.
     * 
     * @param {string} paramName - Parameter name to get from the URL.
     */
   /* getUrlParameter: function (paramName) {
        let url = window.location.href;
        paramName = paramName.replace(/[\[\]]/g, "\\$&");
        var regex = new RegExp("[?&]" + paramName + "(=([^&#]*)|&|#|$)"), results = regex.exec(url);
        if (!results) return null;
        if (!results[2]) return '';
        return decodeURIComponent(results[2].replace(/\+/g, " "));
    },*/
    /*optimizedResize: (function () {
        
            var callbacks = [],
                running = false;
        
            // fired on resize event
            function resize() {
        
                if (!running) {
                    running = true;
        
                    if (window.requestAnimationFrame) {
                        window.requestAnimationFrame(runCallbacks);
                    } else {
                        setTimeout(runCallbacks, 66);
                    }
                }
        
            }
        
            // run the actual callbacks
            function runCallbacks() {
        
                callbacks.forEach(function (callback) {
                    callback.method(callback.params);
                });
        
                running = false;
            }
        
            // adds callback to loop
            function addCallback(callback) {
                if (callback) {
                    callbacks.push(callback);
                }
            }

            function removeCallBack(callback) {
                callbacks = [];
            }
        
            return {
                // public method to add additional callback
                add: (callback) => {
                    if (!callbacks.length) {
                        window.addEventListener('resize', resize);
                    }
                    addCallback(callback);
                },
                remove: () => {
                    removeCallBack();
                    window.removeEventListener('resize', resize);
                }
            }
        }())*/
}