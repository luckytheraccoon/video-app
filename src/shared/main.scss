
@import 'sassPartials/functions';
@import 'sassPartials/mixins';

@import 'sassPartials/grids/singleColGrid';
@import 'sassPartials/grids/twoColGrid';
@import 'sassPartials/grids/threeColGrid';

html {
	overflow-x: hidden;
}

html, body {
	margin:0;
	padding:0;
	font: {
		family: 'Varela Round', sans-serif;
		size:14px;
	}
	@include full-span();
}

#app {
	margin:0 auto;
	max-width:1000px;
	min-width:260px;
	&>div {
		@include full-span();
	}
}

a {
	cursor:pointer;
	color: #317795;
	&:link, &:visited {
	    text-decoration: none !important;
	    font-weight: normal;
	}

	&:hover, &:focus {
	    color: #23527c;
	}

}

.form--search-form {
	grid-area: searchbar;
    flex: 1 100%;
	justify-content: space-between;
	flex-wrap: wrap;
	align-self: start;
	&>input.search-box {
		flex-grow: 1;
		margin-right:2px;
	}
	&>input.submit-button {
		flex-shrink: 1;
	}
}

.div--user-block {
	text-align:center;
	&>.div--user-photo {
		font-size: 5em;
		line-height: 0.1em;
		padding: 10px;
	}
}

.div--main-menu {
	grid-area: maincontent-left;  
	background: #f1f1f1;
    z-index: 1;
    flex-direction: column;
    justify-content: space-between;
	min-width: 150px;
	&>div {
		flex-shrink: 1;
	}

	&>.div--main-menu-item-wrapper {
		flex-grow: 1;
		padding:0 5px 5px 5px;
		&>button {
			width: 100%;
			margin-top:5px;
		}
	}

	&>.div--main-menu-logout {
		padding:5px;
		&>button {
			width: 100%;
		}
	}
}

.div--video-list {
	&>.div--video-list-wrapper {
		align-self: start;
		grid-area: maincontent;
	}
}

.div--video-detail,
.div--video-list {
	&>.div--video-list-wrapper {
		position:relative;
		grid-area: maincontent-right;

		&>.div--video-wrapper-inner {
			display: flex;
			flex-flow:wrap;

			&>.div--video-inner-item {
				margin:0 0 10px 0;

				&>.div--video-title {
					font-size:0.9em;
					margin:0;
				}
				&>.div--video-description {
					font-size:0.8em;
				}
			}
		}
	}
}


.div--footer {
	grid-area: footer;
	display: flex;
	justify-content: space-between;
	flex-wrap: wrap;
	background-color:#f1f1f1;
	padding:25px;
	color:#888;
	&>div {
		margin:5px;
		flex-shrink:1;
	}
	&>div:last-child {
		text-align: right;
		flex-grow:1;
	}
}

.iframe--video {
	position:absolute;
}

.div--video-thumbnail>a>img {
	width:100%;
	max-width: 390px;
	cursor: pointer;
}


.glyphicon {
	cursor: pointer;
}
a>.glyphicon {
	color:white;
}

.div--more-results-button,
.div--go-back-button,
.div--more-results-button-no-click {
	width: 100%!important;
	padding: 10px!important;
	background: #f1f1f1;
	border:1px solid #CCC;
	text-align: center;
	cursor: pointer;
	color:black;

	&>.glyphicon {
		color:black;
	} 

	&:hover {
		border:1px solid #999;
		background: #f5f5f5;
	}

}

.div--more-results-button-no-click {
	&:hover {
		background: #f1f1f1;
		border:1px solid #f1f1f1;
	}
}

@import 'sassPartials/devices/commonScreens';
@import 'sassPartials/devices/smallLaptops';
@import 'sassPartials/devices/tablets';
@import 'sassPartials/devices/phablets';
@import 'sassPartials/devices/smartphones';