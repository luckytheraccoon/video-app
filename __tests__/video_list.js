import React from "react";
import ReactDOM from "react-dom";
import {MainContainer} from '../raw_resources/js/src/components/main';
import {VideoList} from '../raw_resources/js/src/components/common';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

test('Video lists render correctly', () => {

  const wrapper = shallow(<MainContainer />); 
  let pr = wrapper.instance().moreResultsAction();
  
  const array = [0,1,2];

  const props = {
    loadMoreAction: jest.fn(),
    contentArray: array
  };
  

   const comp = shallow(<VideoList {...props} />); 

  expect(toJson(comp)).toMatchSnapshot();
});


