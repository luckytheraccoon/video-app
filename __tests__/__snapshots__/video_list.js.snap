// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video lists render correctly 1`] = `
<div
  className="div-content-list"
>
  <ContentItemList
    pageId="video-list"
  />
  <ContentItemList
    pageId="video-list"
  />
  <ContentItemList
    pageId="video-list"
  />
  <LoadMoreButton />
</div>
`;
